pipeline_dp/dp_engine.py

```diff
@@ -57,6 +46,57 @@ def _add_report_stages(self, stages_description):
         for stage_description in stages_description:
             self._add_report_stage(stage_description)
 
+    def calculate_private_contribution_bounds(
```

Comment:
Could you please move it as a last method of DPEngine?

Reply:
Done.